mobile/apps/auth/lib/services/local_backup_service.dart

```diff
@@ -0,0 +1,45 @@
+import 'dart:io'; 
+
+import 'package:ente_auth/store/code_store.dart';
+import 'package:flutter/foundation.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+class LocalBackupService {
+  static final LocalBackupService instance =
+      LocalBackupService._privateConstructor();
+  LocalBackupService._privateConstructor();
+
+  Future<void> init() async {}
+
+  Future<void> triggerAutomaticBackup() async {
+    try {
+      final prefs = await SharedPreferences.getInstance();
+
+      final isEnabled = prefs.getBool('isAutoBackupEnabled') ?? false;    //checks if toggle is on
+      if (!isEnabled) return;
+
+      final backupPath = prefs.getString('autoBackupPath');
+      if (backupPath == null) return;
+
+      debugPrint("--- Change detected, triggering automatic backup... ---");
+
+      final allCodes = await CodeStore.instance.getAllCodes(sortCodes: false);
+      final validCodes = allCodes.where((code) => !code.hasError);
+      String backupContent = "";
+      for (final code in validCodes) {
+        backupContent += "${code.toOTPAuthUrlFormat()}\n";
+      }
+
+      if (backupContent.trim().isEmpty) return;
+
+      const fileName = 'ente-auth-auto-backup.txt';
```

Comment:
We should include date in the file name?

Also, should we automatically cleanup exports older than x days?

Can we check how other similar app manages the lifecycle for export files just to get some idea about potential expectation from the users. May be in the initially cut we don't really need to auto delete..